docs/spec/auth/vesting.md

```diff
@@ -0,0 +1,162 @@
+## Vesting
+
+### Intro and Requirements
+
+This paper specifies vesting account implementation for the Cosmos Hub. 
+The requirements for this vesting account is that it should be initialized during genesis with
+a starting balance X coins and a vesting endtime T. The owner of this account should be able to delegate to validators 
+and vote with locked coins, however they cannot send locked coins to other accounts until those coins have been unlocked. 
+The vesting account should also be able to spend any coins it receives from other users. 
+Thus, the bank module's `MsgSend` handler should error if a vesting account is trying to send an amount that exceeds their 
+unlocked coin amount.
+
+### Implementation
+
+##### Vesting Account implementation
+
+NOTE:  `Now = ctx.BlockHeader().Time`
+
+```go
+type VestingAccount interface {
+    Account
+    AssertIsVestingAccount() // existence implies that account is vesting.
+    ConvertAccount(sdk.Context) BaseAccount
+
+    // Calculates total amount of unlocked coins released by vesting schedule
+    // May be larger than total coins in account right now
+    TotalUnlockedCoins(sdk.Context) sdk.Coins
+}
+
+// Implements Vesting Account
+// Continuously vests by unlocking coins linearly with respect to time
+type ContinuousVestingAccount struct {
+    BaseAccount
+    OriginalCoins sdk.Coins // Coins in account on Initialization
+    ReceivedCoins sdk.Coins // Coins received from other accounts
+
+    // StartTime and EndTime used to calculate how much of OriginalCoins is unlocked at any given point
+    StartTime     int64
+    EndTime       int64
+}
+
+// ConvertAccount converts VestingAccount into BaseAccount
+// Will convert only after account has fully vested
+ConvertAccount(vacc ContinuousVestingAccount, ctx sdk.Context) (BaseAccount):
+    if Now > vacc.EndTime:
+        return vacc.BaseAccount
+
+// Uses time in context to calculate total unlocked coins
+TotalUnlockedCoins(vacc ContinuousVestingAccount, ctx sdk.Context) sdk.Coins:
+    unlockedCoins := ReceivedCoins + OriginalCoins * (Now - StartTime) / (EndTime - StartTime)
```

Comment:
Yeah for now it's continuous because that is what jae and I discussed last. I'm currently trying to confirm what the preferred vesting schedule is for AiB.

Changing the vesting schedule just comes down to changing implementation of this single method.